app_flutter/lib/logic/task_grid_filter.dart

```diff
@@ -0,0 +1,232 @@
+/// stage type toggles.
+class TaskGridFilter extends FilterPropertySource {
+  TaskGridFilter();
+
+  factory TaskGridFilter.fromMap(Map<String, String> valueMap) => TaskGridFilter()..applyMap(valueMap);
+
+  /// True iff all of the properties of the filter are set to their default values.
+  bool get isDefault =>
+      _allProperties.values.toList().every((ValueFilterProperty<dynamic> element) => element.isDefault);
+
+  /// Sets all properties of this filter to their default values.
+  void reset() {
+    for (final ValueFilterProperty<dynamic> property in _allProperties.values) {
+      property.reset();
+    }
+  }
+
+  /// Modifies this filter based on the values in the map. If the map is null, no changes are
+  /// made. All keys in the [valueMap] must match one of the field names of the filter properties.
+  void applyMap(Map<String, String> valueMap) {
+    if (valueMap != null) {
+      for (final MapEntry<String, String> mapEntry in valueMap.entries) {
+        _allProperties[mapEntry.key].stringValue = mapEntry.value;
+      }
+    }
+  }
+
+  final RegExpFilterProperty _taskProperty = RegExpFilterProperty(fieldName: 'taskFilter', label: 'Task Name');
+  final RegExpFilterProperty _authorProperty = RegExpFilterProperty(fieldName: 'authorFilter', label: 'Commit Author');
+  final RegExpFilterProperty _messageProperty =
+      RegExpFilterProperty(fieldName: 'messageFilter', label: 'Commit Message');
+  final RegExpFilterProperty _hashProperty = RegExpFilterProperty(fieldName: 'hashFilter', label: 'Commit Hash');
+  final BoolFilterProperty _androidProperty = BoolFilterProperty(fieldName: 'showAndroid', label: 'Android');
+  final BoolFilterProperty _iosProperty = BoolFilterProperty(fieldName: 'showIos', label: 'iOS');
+  final BoolFilterProperty _windowsProperty = BoolFilterProperty(fieldName: 'showWindows', label: 'Windows');
+  final BoolFilterProperty _cirrusProperty = BoolFilterProperty(fieldName: 'showCirrus', label: 'Cirrus');
+  final BoolFilterProperty _luciProperty = BoolFilterProperty(fieldName: 'showLuci', label: 'Luci');
+
+  // [_allProperties] is a LinkedHashMap so we can trust its iteration order
+  LinkedHashMap<String, ValueFilterProperty<dynamic>> _allPropertiesMap;
+  LinkedHashMap<String, ValueFilterProperty<dynamic>> get _allProperties =>
+      _allPropertiesMap ??= <String, ValueFilterProperty<dynamic>>{}
+        ..[_taskProperty.fieldName] = _taskProperty
+        ..[_authorProperty.fieldName] = _authorProperty
+        ..[_messageProperty.fieldName] = _messageProperty
+        ..[_hashProperty.fieldName] = _hashProperty
+        ..[_androidProperty.fieldName] = _androidProperty
+        ..[_iosProperty.fieldName] = _iosProperty
+        ..[_windowsProperty.fieldName] = _windowsProperty
+        ..[_cirrusProperty.fieldName] = _cirrusProperty
+        ..[_luciProperty.fieldName] = _luciProperty;
+
+  // TODO(flar): Decide whether or not to expose these fields or do everything through the
```

Comment:
Removed commented out code?

Reply:
There is an implicit question there, but I take it you side on the "not" part of whether these accessors make sense?

Reply:
I ended up putting them in (uncommenting this section) as they helped a lot for testing.